sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java

```diff
@@ -530,4 +532,74 @@ public void testFipsEnabled() {
         kubernetesCluster.refresh();
         Assertions.assertTrue(kubernetesCluster.agentPools().get(agentPoolName2).isFipsEnabled());
     }
+
+    @Test
+    public void testUpdateVersionFromDefaultToSpecified() {
+        resourceManager.resourceGroups().define(rgName).withRegion(Region.US_EAST).create();
+        KubernetesCluster kubernetesCluster = containerServiceManager.kubernetesClusters()
+            .define(generateRandomResourceName("aks", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withDefaultVersion()
+            .withRootUsername("testaks")
+            .withSshKey(SSH_KEY)
+            .withSystemAssignedManagedServiceIdentity()
+            .defineAgentPool(generateRandomResourceName("ap", 15))
+            .withVirtualMachineSize(ContainerServiceVMSizeTypes.STANDARD_D2_V2)
+            .withAgentPoolVirtualMachineCount(1)
+            .withAgentPoolType(AgentPoolType.VIRTUAL_MACHINE_SCALE_SETS)
+            .withAgentPoolMode(AgentPoolMode.SYSTEM)
+            .attach()
+            .withDnsPrefix(generateRandomResourceName("dns", 15))
+            .create();
+
+        kubernetesCluster.refresh();
+        Assertions.assertEquals("1.27", kubernetesCluster.version());
```

Comment:
Default version is not fixed I guess? We probably don't want to assert that. @weidongxu-microsoft

Reply:
Yes. Agree. Though it may be hard to do the assert without the exact version.

Reply:
We could split this test into two parts?
1. create withDefaultVersion, assert version not null
2. create with exact version(1.27) and upgrade it to newer(1.28.5)

Reply:
Yes, this maybe better.

(though it still be possible that after a year AKS decide not support 1.27 -- but that still better than now)

Reply:
Added test case `KubernetesClustersTests.canCreateClusterWithDefaultVersion` in the new version.